native/packages/react-native-bpk-component-button/src/utils.js

```diff
@@ -79,23 +82,27 @@ export const getStyleForElement = (
 };
 
 export const getThemingForElement = (
-  elementType,
-  theme,
-  { type, disabled },
-) => {
+  elementType: string,
+  theme: ?Theme,
+  { type, disabled }: Object,
+): Object => {
+  const suppliedTheme = theme; // This is purely to stop Flow from compaining.
```

Comment:
Why does Flow complain without this?

Reply:
It's the thing we talked about yesterday - it complains that `theme` could potentially be undefined:

![screen shot 2018-01-11 at 16 20 15](https://user-images.githubusercontent.com/73652/34834985-6cb4ec58-f6eb-11e7-97b4-17fc5255341a.png)

I tried a few ways to reorganise the code to make it not complain, and the only thing that worked for me was moving the check that `theme` is defined to directly above where it's used:

![screen shot 2018-01-11 at 16 20 46](https://user-images.githubusercontent.com/73652/34835008-7fc18a36-f6eb-11e7-893e-78599a7c9063.png)

But that feels to me like I'm making the code worse, as it's doing that check multiple times now, where it only needs to be done once. So I went with this approach to keep the code efficient.

What do you reckon?

Reply:
Fair I just think it's strange since the 

```
if (theme && ...)
```

Should refine the type inside the if from `?Theme` to `Theme`. Maybe Flow cannot understand complex expressions when refining types? I.e if we used

```
if (theme) {
  if (styles.themeMappings[elementType]) {
    // ...
```

It might be happier

Reply:
I think I tried that, but I'll try it again shortly and report back.

Reply:
![screen shot 2018-01-12 at 11 04 42](https://user-images.githubusercontent.com/73652/34872362-7a32e95a-f788-11e7-92d3-e484b6f5c810.png)

`¯\_(ツ)_/¯`